server/graphql/v2/query/ActivityTimelineQuery.ts

```diff
@@ -0,0 +1,70 @@
+import assert from 'assert';
+
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+import { InferAttributes, Order, WhereOptions } from 'sequelize';
+
+import ActivityTypes, { ActivitiesPerClass, ActivityClasses } from '../../../constants/activities';
+import { types as AccountTypes } from '../../../constants/collectives';
+import models, { Op } from '../../../models';
+import Activity from '../../../models/Activity';
+import { checkRemoteUserCanUseAccount } from '../../common/scope-check';
+import { Unauthorized } from '../../errors';
+import { GraphQLActivityCollection } from '../collection/ActivityCollection';
+import { idDecode } from '../identifiers';
+import { CollectionArgs } from '../interface/Collection';
+
+const generateQuery = (account): WhereOptions<InferAttributes<Activity, { omit: never }>> => {
+  if (account.type === AccountTypes.USER) {
+    return {
+      type: { [Op.notIn]: [ActivityTypes.COLLECTIVE_TRANSACTION_CREATED, ActivityTypes.ORDER_PENDING_CREATED] },
+      [Op.or]: [
+        { UserId: account.CreatedByUserId },
+        // Events on expenses the user submitted
+        { type: { [Op.in]: ActivitiesPerClass[ActivityClasses.EXPENSES] }, FromCollectiveId: account.id },
+      ],
+    };
+  } else {
+    return { [Op.or]: [{ CollectiveId: account.id }, { FromCollectiveId: account.id }] };
+  }
+};
+
+const ActivityTimelineQuery = {
+  type: new GraphQLNonNull(GraphQLActivityCollection),
+  description: "Activity timeline for the account's workspace",
+  args: {
+    id: {
+      type: GraphQLString,
+      description: `The public id identifying the account (ie: dgm9bnk8-0437xqry-ejpvzeol-jdayw5re)`,
+    },
+    slug: {
+      type: GraphQLString,
+      description: `The slug identifying the account (ie: babel for https://opencollective.com/babel)`,
+    },
```

Comment:
We could use `account: GraphQLAccountReferenceInput` instead. That way you have access to `fetchAccountWithReference` to easily get the account.